src/unstructured_client/utils/_human_split_pdf.py

```diff
@@ -0,0 +1,137 @@
+import copy
+import io
+import logging
+import os
+import functools
+from typing import Optional, Tuple, Callable
+from concurrent.futures import ThreadPoolExecutor
+
+from pypdf import PdfReader, PdfWriter
+
+from unstructured_client import utils
+from unstructured_client.models import shared, operations
+
+logger = logging.getLogger('unstructured-client')
+
+
+def handle_split_pdf_page(func: Callable) -> Callable:
+    """
+    A decorator for splitting PDF by pages and sending each of them separately to backend.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> operations.PartitionResponse:
+        if len(args) > 0:
+            request = args[1]
```

Comment:
Add handling for if users define `request` using a kwarg instead of a positional arg, similar to the `clean_server_url` decorator:

```
server_url: Optional[str] = cast(Optional[str], kwargs.get("server_url"))

if server_url is None and len(args) > SERVER_URL_ARG_IDX:
    server_url = cast(str, args[SERVER_URL_ARG_IDX])
    url_is_in_kwargs = False
```

Reply:
Done.